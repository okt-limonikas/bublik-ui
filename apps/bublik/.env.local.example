# Username and password used to proxy requests to JSON logs
BUBLIK_UI_DEV_LOGS_AUTH="<USERNAME>:<PASSWORD>"
# Target of JSON logs (protocol, host, port)
BUBLIK_UI_DEV_LOGS_TARGET="https://example.com"
# Where backend is served (protocol, host, port)
BUBLIK_UI_DEV_BACKEND_TARGET="http://localhost:8000"
