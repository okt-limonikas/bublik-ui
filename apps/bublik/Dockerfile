FROM node:17.9.0

ENV PNPM_HOME="/root/.local/share/pnpm"
ENV PATH="${PATH}:${PNPM_HOME}"

RUN npm install --global pnpm

RUN pnpm add -g nx

WORKDIR /app

COPY package.json pnpm-lock.yaml ./
RUN pnpm install

COPY . .

CMD ["pnpm", "build", "bublik", "--prod"]
