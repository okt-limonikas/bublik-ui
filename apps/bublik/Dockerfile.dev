FROM node:22.13-alpine AS base

ARG PLAYWRIGHT_SKIP_BROWSER_DOWNLOAD=1

ENV PNPM_HOME="/pnpm"
ENV PATH="$PNPM_HOME:$PATH"

ENV PLAYWRIGHT_SKIP_BROWSER_DOWNLOAD=${PLAYWRIGHT_SKIP_BROWSER_DOWNLOAD}
ENV BASE_URL="/v2"

RUN npm i -g corepack@latest
RUN corepack enable

COPY . /app
WORKDIR /app

FROM base AS runner

RUN --mount=type=cache,id=pnpm,target=/pnpm/store pnpm install --frozen-lockfile --ignore-scripts

EXPOSE 4200

CMD pnpm run nx serve --host=0.0.0.0 --base=${BASE_URL}
