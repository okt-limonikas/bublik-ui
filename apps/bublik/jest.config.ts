/* SPDX-License-Identifier: Apache-2.0 */
/* SPDX-FileCopyrightText: 2021-2023 OKTET Labs Ltd. */
/* eslint-disable */
export default {
	displayName: 'bublik',
	preset: '../../jest.preset.js',
	transform: {
		'^(?!.*\\.(js|jsx|ts|tsx|css|json)$)': '@nx/react/plugins/jest',
		'^.+\\.[tj]sx?$': 'babel-jest'
	},
	moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx'],
	coverageDirectory: '../../coverage/apps/bublik'
};
