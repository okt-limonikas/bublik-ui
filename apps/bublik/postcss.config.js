/* SPDX-License-Identifier: Apache-2.0 */
/* SPDX-FileCopyrightText: 2021-2023 OKTET Labs Ltd. */
const { join } = require('path');

module.exports = {
	plugins: {
		tailwindcss: { config: join(__dirname, 'tailwind.config.js') },
		autoprefixer: {}
	}
};
