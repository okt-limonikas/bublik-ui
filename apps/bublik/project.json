{
	"name": "bublik",
	"$schema": "../../node_modules/nx/schemas/project-schema.json",
	"sourceRoot": "apps/bublik/src",
	"projectType": "application",
	"tags": [],
	"targets": {
		"build": {
			"executor": "@nx/vite:build",
			"outputs": ["{options.outputPath}"],
			"options": {
				"outputPath": "dist/apps/bublik"
			},
			"configurations": {
				"production": {
					"base": "/v2",
					"sourcemap": false
				}
			}
		},
		"serve": {
			"executor": "@nx/vite:dev-server",
			"options": {
				"buildTarget": "bublik:build",
				"hmr": true
			},
			"configurations": {}
		},
		"lint": {
			"executor": "@nx/eslint:lint"
		},
		"test": {
			"executor": "@nx/vite:test",
			"outputs": ["{workspaceRoot}/coverage/apps/bublik"],
			"options": {
				"passWithNoTests": true
			}
		},
		"storybook": {
			"options": {
				"port": 4400,
				"config-dir": ".storybook"
			},
			"configurations": {
				"ci": {
					"args": ["--quiet"]
				}
			}
		},
		"build-storybook": {
			"outputs": [
				"{projectRoot}/{options.output-dir}",
				"{workspaceRoot}/{projectRoot}/storybook-static",
				"{options.output-dir}",
				"{options.outputDir}",
				"{options.o}"
			],
			"options": {
				"config-dir": ".storybook",
				"output-dir": "../../dist/storybook/bublik"
			},
			"configurations": {
				"ci": {
					"quiet": true
				}
			}
		},
		"test-storybook": {
			"executor": "nx:run-commands",
			"options": {
				"command": "test-storybook -c apps/bublik/.storybook --url=http://localhost:4400"
			}
		},
		"e2e": {
			"executor": "@nx/playwright:playwright",
			"outputs": ["{workspaceRoot}/dist/.playwright/apps/bublik"],
			"options": {
				"config": "apps/bublik/playwright.config.ts"
			}
		}
	}
}
