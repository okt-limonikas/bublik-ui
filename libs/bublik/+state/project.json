{
	"name": "bublik-+state",
	"$schema": "../../../node_modules/nx/schemas/project-schema.json",
	"sourceRoot": "libs/bublik/+state/src",
	"projectType": "library",
	"targets": {
		"lint": {
			"executor": "@nx/eslint:lint",
			"options": {
				"lintFilePatterns": ["libs/bublik/+state/**/*.{ts,tsx,js,jsx}"]
			}
		},
		"test": {
			"executor": "@nx/jest:jest",
			"outputs": ["{workspaceRoot}/coverage/libs/bublik/+state"],
			"options": {
				"jestConfig": "libs/bublik/+state/jest.config.ts",
				"passWithNoTests": true
			}
		}
	},
	"tags": []
}
