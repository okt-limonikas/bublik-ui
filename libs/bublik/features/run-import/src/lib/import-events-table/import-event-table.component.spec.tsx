/* SPDX-License-Identifier: Apache-2.0 */
/* SPDX-FileCopyrightText: 2021-2023 OKTET Labs Ltd. */
import { useState } from 'react';
import { PaginationState } from '@tanstack/react-table';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';

import { store } from '@/bublik/+state';

import {
	ImportEventTable,
	ImportEventTableEmpty,
	ImportEventTableError,
	ImportEventTableLoading
} from './import-event-table.component';

function EventTableDemo() {
	const [pagination, setPagination] = useState<PaginationState>({
		pageIndex: 0,
		pageSize: 25
	});

	return (
		<Provider store={store}>
			<ImportEventTable
				data={[]}
				pagination={pagination}
				setPagination={setPagination}
			/>
		</Provider>
	);
}

describe('ImportEventTableComponent', () => {
	it('should render successfully', () => {
		const { asFragment } = render(<EventTableDemo />);

		expect(asFragment()).toMatchSnapshot();
	});

	it('should render loading state', () => {
		const { asFragment } = render(<ImportEventTableLoading />);

		expect(asFragment()).toMatchSnapshot();
	});

	it('should render error state', () => {
		const { asFragment } = render(
			<ImportEventTableError error={{ status: 404 }} />
		);

		expect(asFragment()).toMatchSnapshot();
	});

	it('should render empty state', () => {
		const { asFragment } = render(<ImportEventTableEmpty />);

		expect(asFragment()).toMatchSnapshot();
	});
});
