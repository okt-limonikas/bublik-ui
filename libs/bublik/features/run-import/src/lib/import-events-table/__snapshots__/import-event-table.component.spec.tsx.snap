// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ImportEventTableComponent > should render empty state 1`] = `
<DocumentFragment>
  <div
    class="grid place-items-center h-[calc(100vh-176px)]"
  >
    <div
      class="flex flex-col items-center text-center"
    >
      <svg
        class="text-text-unexpected"
        fill="none"
        height="48"
        viewBox="0 0 17 16"
        width="48"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M15.4346 11.4571C15.4459 11.4765 15.4553 11.4971 15.4639 11.5178C15.5433 11.7131 15.5886 11.9191 15.5999 12.1298C15.6259 12.6331 15.4546 13.1158 15.1166 13.4905C14.7793 13.8651 14.3159 14.0858 13.8126 14.1111L6.64126 14.1118C6.36526 14.1118 6.14126 13.8878 6.14126 13.6118C6.14126 13.3358 6.36526 13.1118 6.64126 13.1118H13.7873C13.9979 13.0998 14.2153 12.9971 14.3746 12.8205C14.5333 12.6445 14.6133 12.4185 14.6013 12.1818C14.5966 12.0925 14.5786 12.0045 14.5473 11.9205L9.7146 3.46313C9.47526 3.03046 8.9346 2.87446 8.5066 3.11113C8.3606 3.19179 8.24126 3.31113 8.1606 3.45713L3.32393 11.9211C3.24593 12.1331 3.25126 12.3631 3.33926 12.5718C3.43126 12.7898 3.6026 12.9591 3.82193 13.0478C3.9146 13.0858 4.0106 13.1071 4.10993 13.1125C4.38526 13.1258 4.59793 13.3611 4.58393 13.6371C4.5706 13.9038 4.34926 14.1118 4.0846 14.1118H4.08456C4.07591 14.1118 4.06792 14.1118 4.05926 14.1111C3.84726 14.1005 3.6406 14.0545 3.4446 13.9745C2.97926 13.7851 2.6146 13.4251 2.4186 12.9611C2.22193 12.4971 2.2186 11.9838 2.4086 11.5171C2.4166 11.4965 2.42593 11.4765 2.43726 11.4571L7.2886 2.96713C7.4566 2.66313 7.71193 2.40846 8.02193 2.23579C8.93326 1.73179 10.0833 2.06246 10.5866 2.97313L15.4346 11.4571ZM8.93594 9.06532C9.21194 9.06532 9.43594 8.84132 9.43594 8.56532V6.68066C9.43594 6.40466 9.21194 6.18066 8.93594 6.18066C8.65994 6.18066 8.43594 6.40466 8.43594 6.68066V8.56532C8.43594 8.84132 8.65994 9.06532 8.93594 9.06532ZM8.93594 11.346C9.21194 11.346 9.43594 11.122 9.43594 10.846V10.8393C9.43594 10.5633 9.21194 10.3427 8.93594 10.3427C8.65994 10.3427 8.43594 10.57 8.43594 10.846C8.43594 11.122 8.65994 11.346 8.93594 11.346Z"
          fill="currentColor"
          fill-rule="evenodd"
        />
      </svg>
      <h3
        class="mt-2 text-sm font-medium text-gray-900"
      >
        No results found
      </h3>
      <p
        class="mt-1 text-sm text-gray-500"
      >
        No results found, please try another search.
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`ImportEventTableComponent > should render error state 1`] = `
<DocumentFragment>
  <div
    class="flex items-center justify-center flex-grow h-screen"
  >
    <div
      class="flex items-center gap-4"
    >
      <svg
        class="text-text-unexpected"
        fill="none"
        height="48"
        viewBox="0 0 17 16"
        width="48"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M15.4346 11.4571C15.4459 11.4765 15.4553 11.4971 15.4639 11.5178C15.5433 11.7131 15.5886 11.9191 15.5999 12.1298C15.6259 12.6331 15.4546 13.1158 15.1166 13.4905C14.7793 13.8651 14.3159 14.0858 13.8126 14.1111L6.64126 14.1118C6.36526 14.1118 6.14126 13.8878 6.14126 13.6118C6.14126 13.3358 6.36526 13.1118 6.64126 13.1118H13.7873C13.9979 13.0998 14.2153 12.9971 14.3746 12.8205C14.5333 12.6445 14.6133 12.4185 14.6013 12.1818C14.5966 12.0925 14.5786 12.0045 14.5473 11.9205L9.7146 3.46313C9.47526 3.03046 8.9346 2.87446 8.5066 3.11113C8.3606 3.19179 8.24126 3.31113 8.1606 3.45713L3.32393 11.9211C3.24593 12.1331 3.25126 12.3631 3.33926 12.5718C3.43126 12.7898 3.6026 12.9591 3.82193 13.0478C3.9146 13.0858 4.0106 13.1071 4.10993 13.1125C4.38526 13.1258 4.59793 13.3611 4.58393 13.6371C4.5706 13.9038 4.34926 14.1118 4.0846 14.1118H4.08456C4.07591 14.1118 4.06792 14.1118 4.05926 14.1111C3.84726 14.1005 3.6406 14.0545 3.4446 13.9745C2.97926 13.7851 2.6146 13.4251 2.4186 12.9611C2.22193 12.4971 2.2186 11.9838 2.4086 11.5171C2.4166 11.4965 2.42593 11.4765 2.43726 11.4571L7.2886 2.96713C7.4566 2.66313 7.71193 2.40846 8.02193 2.23579C8.93326 1.73179 10.0833 2.06246 10.5866 2.97313L15.4346 11.4571ZM8.93594 9.06532C9.21194 9.06532 9.43594 8.84132 9.43594 8.56532V6.68066C9.43594 6.40466 9.21194 6.18066 8.93594 6.18066C8.65994 6.18066 8.43594 6.40466 8.43594 6.68066V8.56532C8.43594 8.84132 8.65994 9.06532 8.93594 9.06532ZM8.93594 11.346C9.21194 11.346 9.43594 11.122 9.43594 10.846V10.8393C9.43594 10.5633 9.21194 10.3427 8.93594 10.3427C8.65994 10.3427 8.43594 10.57 8.43594 10.846C8.43594 11.122 8.65994 11.346 8.93594 11.346Z"
          fill="currentColor"
          fill-rule="evenodd"
        />
      </svg>
      <div
        class="flex flex-col gap-0.5"
      >
        <h2
          class="text-2xl font-bold"
        >
          404 Not found
        </h2>
        <p
          class="text-lg"
        >
          The requested resource could not be found
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`ImportEventTableComponent > should render loading state 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col gap-1 mt-1"
  >
    <div
      class="animate-pulse bg-gray-300 h-10"
    />
    <div
      class="animate-pulse bg-gray-300 rounded h-[52.5px]"
    />
    <div
      class="animate-pulse bg-gray-300 rounded h-[52.5px]"
    />
    <div
      class="animate-pulse bg-gray-300 rounded h-[52.5px]"
    />
    <div
      class="animate-pulse bg-gray-300 rounded h-[52.5px]"
    />
    <div
      class="animate-pulse bg-gray-300 rounded h-[52.5px]"
    />
    <div
      class="animate-pulse bg-gray-300 rounded h-[52.5px]"
    />
    <div
      class="animate-pulse bg-gray-300 rounded h-[52.5px]"
    />
    <div
      class="animate-pulse bg-gray-300 rounded h-[52.5px]"
    />
    <div
      class="animate-pulse bg-gray-300 rounded h-[52.5px]"
    />
    <div
      class="animate-pulse bg-gray-300 rounded h-[52.5px]"
    />
    <div
      class="animate-pulse bg-gray-300 rounded h-[52.5px]"
    />
    <div
      class="animate-pulse bg-gray-300 rounded h-[52.5px]"
    />
    <div
      class="animate-pulse bg-gray-300 rounded h-[52.5px]"
    />
    <div
      class="animate-pulse bg-gray-300 rounded h-[52.5px]"
    />
    <div
      class="animate-pulse bg-gray-300 rounded h-[52.5px]"
    />
    <div
      class="animate-pulse bg-gray-300 rounded h-[52.5px]"
    />
    <div
      class="animate-pulse bg-gray-300 rounded h-[52.5px]"
    />
    <div
      class="animate-pulse bg-gray-300 rounded h-[52.5px]"
    />
    <div
      class="animate-pulse bg-gray-300 rounded h-[52.5px]"
    />
    <div
      class="animate-pulse bg-gray-300 rounded h-[52.5px]"
    />
    <div
      class="animate-pulse bg-gray-300 rounded h-[52.5px]"
    />
    <div
      class="animate-pulse bg-gray-300 rounded h-[52.5px]"
    />
    <div
      class="animate-pulse bg-gray-300 rounded h-[52.5px]"
    />
    <div
      class="animate-pulse bg-gray-300 rounded h-[52.5px]"
    />
    <div
      class="animate-pulse bg-gray-300 rounded h-[52.5px]"
    />
  </div>
</DocumentFragment>
`;

exports[`ImportEventTableComponent > should render successfully 1`] = `
<DocumentFragment>
  <div>
    <table
      class="border-separate border-spacing-y-1 w-full h-full"
    >
      <thead
        class="bg-white sticky top-0 z-10 transition-shadow"
      >
        <tr
          class="h-8.5"
        >
          <th
            class="tracking-wider text-left text-[0.6875rem] font-semibold leading-[0.875rem] w-px p-0"
            colspan="1"
          />
          <th
            class="px-1 py-2 tracking-wider text-left text-[0.6875rem] font-semibold leading-[0.875rem] w-px"
            colspan="1"
          >
            <span
              class="pl-2"
            >
              Actions
            </span>
          </th>
          <th
            class="px-1 py-2 tracking-wider text-left text-[0.6875rem] font-semibold leading-[0.875rem]"
            colspan="1"
          >
            Date
          </th>
          <th
            class="px-1 py-2 tracking-wider text-left text-[0.6875rem] font-semibold leading-[0.875rem]"
            colspan="1"
          >
            <span
              class="pl-2"
            >
              Facility
            </span>
          </th>
          <th
            class="px-1 py-2 tracking-wider text-left text-[0.6875rem] font-semibold leading-[0.875rem]"
            colspan="1"
          >
            <span
              class="pl-2"
            >
              Severity
            </span>
          </th>
          <th
            class="px-1 py-2 tracking-wider text-left text-[0.6875rem] font-semibold leading-[0.875rem]"
            colspan="1"
          >
            Message
          </th>
          <th
            class="px-1 py-2 tracking-wider text-left text-[0.6875rem] font-semibold leading-[0.875rem]"
            colspan="1"
          >
            URL
          </th>
        </tr>
      </thead>
      <tbody
        class="bg-white"
      />
    </table>
    <div
      class="flex items-center justify-center"
    >
      <div
        class="flex gap-1"
        data-testid="tw-pagination"
      >
        <button
          class="flex items-center py-2 px-4 rounded-md text-[1rem] hover:bg-primary hover:text-white disabled:text-text-menu disabled:cursor-not-allowed disabled:bg-white disabled:hover:text-text-menu text-text-primary bg-white"
          disabled=""
        >
          Previous
        </button>
        <button
          class="flex items-center py-2 px-4 rounded-md text-[1rem] hover:bg-primary hover:text-white disabled:text-text-menu disabled:cursor-not-allowed disabled:bg-white disabled:hover:text-text-menu bg-primary text-white"
        >
          1
        </button>
        <button
          class="flex items-center py-2 px-4 rounded-md text-[1rem] hover:bg-primary hover:text-white disabled:text-text-menu disabled:cursor-not-allowed disabled:bg-white disabled:hover:text-text-menu text-text-primary bg-white"
        >
          …
        </button>
        <button
          class="flex items-center py-2 px-4 rounded-md text-[1rem] hover:bg-primary hover:text-white disabled:text-text-menu disabled:cursor-not-allowed disabled:bg-white disabled:hover:text-text-menu text-text-primary bg-white"
        >
          …
        </button>
        <button
          class="flex items-center py-2 px-4 rounded-md text-[1rem] hover:bg-primary hover:text-white disabled:text-text-menu disabled:cursor-not-allowed disabled:bg-white disabled:hover:text-text-menu text-text-primary bg-white"
        >
          NaN
        </button>
        <button
          class="flex items-center py-2 px-4 rounded-md text-[1rem] hover:bg-primary hover:text-white disabled:text-text-menu disabled:cursor-not-allowed disabled:bg-white disabled:hover:text-text-menu text-text-primary bg-white"
        >
          Next
        </button>
        <button
          aria-autocomplete="none"
          aria-controls="radix-:r0:"
          aria-expanded="false"
          aria-label="Page sizes"
          class="inline-flex items-center justify-center rounded-md px-4 text-[1rem] font-normal gap-1 bg-white hover:text-primary outline-none"
          data-state="closed"
          dir="ltr"
          role="combobox"
          type="button"
        >
          <span
            style="pointer-events: none;"
          >
            25
          </span>
          <span
            aria-hidden="true"
          >
            <svg
              fill="none"
              height="15"
              viewBox="0 0 15 15"
              width="15"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M3.13523 6.15803C3.3241 5.95657 3.64052 5.94637 3.84197 6.13523L7.5 9.56464L11.158 6.13523C11.3595 5.94637 11.6759 5.95657 11.8648 6.15803C12.0536 6.35949 12.0434 6.67591 11.842 6.86477L7.84197 10.6148C7.64964 10.7951 7.35036 10.7951 7.15803 10.6148L3.15803 6.86477C2.95657 6.67591 2.94637 6.35949 3.13523 6.15803Z"
                fill="currentColor"
                fill-rule="evenodd"
              />
            </svg>
          </span>
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
